desc/magnetic_fields.py

```diff
@@ -861,18 +1018,19 @@ def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
                 self._method,
                 (0, 0, 0),
                 self._extrap,
-                (None, self._period, None),
+                (None, 2 * np.pi / self.NFP, None),
                 **self._derivs["BZ"],
             )
-        B = jnp.array([BRq, Bphiq, BZq]).T
+        # BRq etc shape(nq, ngroups)
+        B = jnp.stack([BRq, Bphiq, BZq], axis=1)
+        # B shape(nq, 3, ngroups)
+        B = jnp.sum(B * currents, axis=-1)
```

Comment:
Why the Ngroups parameter added? So that you can pass in different sources of fields, which may have different scaling factors associated with them, without having to manualy scale/sum them before passing into the class?